ddsketch/ddsketch.go

```diff
@@ -216,3 +216,31 @@ func FromProto(pb *sketchpb.DDSketch) (*DDSketch, error) {
 		zeroCount:          pb.ZeroCount,
 	}, nil
 }
+
+// ChangeMapping changes the store to a new mapping. It is not the conversion that minimizes the loss in relative
+// accuracy, but it avoids artefacts like empty bins that make the histograms look bad.
+// scaleFactor allows to scale out / in all values. (changing units for eg)
+func (s *DDSketch) ChangeMapping(newMapping mapping.IndexMapping, positiveStore store.Store, negativeStore store.Store, scaleFactor float64) *DDSketch {
```

Comment:
Can we make it explicit that `negativeStore` and `positiveStore` are meant to be initially empty and distinct stores? Can we also mention that `s` is unchanged by the operation and that a new distinct sketch is returned?